lib/metrics/context.js

```diff
@@ -104,18 +103,20 @@ module.exports = function (log, config) {
    * @param data target object
    */
   function gather (data) {
-    const metadata = this.payload && this.payload.metricsContext
     let token
 
     return P.resolve()
       .then(() => {
+        const metadata = this.payload && this.payload.metricsContext
```

Comment:
I moved this local to the inner scope to avoid any confusion when a later scope re-uses the same variable name.